app/services/samples/transfer_service.rb

```diff
@@ -37,12 +35,20 @@ def validate(new_project_id, sample_ids)
     end
 
     def transfer(new_project_id, sample_ids)
-      ActiveRecord::Base.transaction do
-        sample_ids.each do |sample_id|
-          sample = Sample.find_by(id: sample_id, project_id: @project.id)
-          sample.update(project_id: new_project_id)
+      transferred_samples_ids = []
+      sample_ids.each do |sample_id|
+        sample = Sample.find_by(id: sample_id, project_id: @project.id)
+        sample.update!(project_id: new_project_id)
+        transferred_samples_ids << sample_id
+      rescue StandardError => e
+        if sample
```

Comment:
Since we already know either the sample will be found or will not we can have specific error messages below. Add new translations under `services` -> `samples` -> `transfers`, and then update the error message portion in the `project.errors.add` for the two errors below to use the translations.

Reply:
Please see 3b999427faedfaea6480c292d4152ba67fff9ac4.

Reply:
Like discussed, please see ffeac9b734edf77af4f179877f73778c837d6441.